tests/D2L.CodeStyle.Analyzers.Test/Async/Generator/AsyncToSyncMethodTransformerTests.cs

```diff
@@ -0,0 +1,104 @@
+﻿using System.Collections.Immutable;
+using D2L.CodeStyle.Analyzers.Async.Generator;
+using D2L.CodeStyle.Annotations;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using NUnit.Framework;
+using static D2L.CodeStyle.Analyzers.Async.Generator.SyntaxTransformer;
+
+namespace D2L.CodeStyle.Analyzers.Tests.Async.Generator;
+
+[TestFixture]
+internal sealed class AsyncToSyncMethodTransformerTests {
+	[Test]
+	public void BasicTask() {
+		var actual = Transform( @"[GenerateSync] async Task HelloAsync( int abc, string def ) { await Task.Delay( 3 ); }" );
+
+		Assert.IsTrue( actual.Success );
+		Assert.IsEmpty( actual.Diagnostics );
+		Assert.AreEqual( "[Blocking] void Hello( int abc, string def ) {throw null;}", actual.Value.ToFullString() );
+	}
+
+	[Test]
+	public void BasicTaskT() {
+		var actual = Transform( @"[GenerateSync] async Task<int> HelloAsync() { await Task.Delay( 3 ); return 4; }" );
+
+		Assert.IsTrue( actual.Success );
+		Assert.IsEmpty( actual.Diagnostics );
+		Assert.AreEqual( "[Blocking] int Hello() {throw null;}", actual.Value.ToFullString() );
+	}
+
+	[Test]
+	public void GenerateSyncOnNonAsyncThingFails() {
+		var actual = Transform( @"[GenerateSync] void Hello() {}" );
+
+		Assert.IsFalse( actual.Success );
+
+		AssertDiagnostics(
+			actual.Diagnostics,
+			// expected:
+			Diagnostics.ExpectedAsyncSuffix,
+			Diagnostics.NonTaskReturnType
+		);
+	}
+
+	// loosly assert that the right sorts of diagnostics came out
+	private static void AssertDiagnostics( IEnumerable<Diagnostic> actual, params DiagnosticDescriptor[] expected ) {
+		CollectionAssert.AreEquivalent(
+			actual.Select( d => d.Descriptor ),
+			expected
+		);
+	}
+
+	public static TransformResult<MethodDeclarationSyntax> Transform(
+		string methodSource,
+		CancellationToken? cancellationToken = null
+	) {
+		var (compilation, methodDecl) = ParseMethod( methodSource );
+
+		var transformer = new AsyncToSyncMethodTransformer(
+			compilation.GetSemanticModel( methodDecl.SyntaxTree ),
+			cancellationToken ?? CancellationToken.None

```

Comment:
So I went to go write tests and unfortunately starting a transformation with a pre-cancelled token didn't do anything. I think because a lot of code paths lead to [this method](https://sourceroslyn.io/#Microsoft.CodeAnalysis.CSharp/Compilation/MemberSemanticModel.cs,9471331dd9be3500,references) which ignores its cancellation token.

I still think it's worth threading through for good form at least, and when the AsyncToSyncMethodTransformer gets bigger there will probably be reasonable places to check manually for cancellation (maybe every block or something?) so I left it in (including this unused support in tests.)